crates/router/src/connector/square/transformers.rs

```diff
@@ -17,19 +17,18 @@ impl TryFrom<(&types::TokenizationRouterData, BankDebitData)> for SquareTokenReq
     fn try_from(
         value: (&types::TokenizationRouterData, BankDebitData),
     ) -> Result<Self, Self::Error> {
-        let (item, bank_debit_data) = value;
+        let (_item, bank_debit_data) = value;
         match bank_debit_data {
             BankDebitData::AchBankDebit { .. } => Err(errors::ConnectorError::NotImplemented(
-                "Payment Method".to_string(),
+                utils::get_unimplemented_payment_method_error_message("Square"),
             ))
             .into_report(),
 
             BankDebitData::SepaBankDebit { .. }
             | BankDebitData::BecsBankDebit { .. }
-            | BankDebitData::BacsBankDebit { .. } => Err(errors::ConnectorError::NotSupported {
-                message: format!("{:?}", item.request.payment_method_data),
-                connector: "Square",
-            })?,
+            | BankDebitData::BacsBankDebit { .. } => Err(errors::ConnectorError::NotImplemented(
+                utils::get_unimplemented_payment_method_error_message("Square"),
```

Comment:
We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice

Reply:
I had assumed the two blocks were kept separate on purpose to have the `AchBank` arm generate a report, and the others not. 

On second look, it seems the conversion happens automatically for the others anyway, to keep the return type identical. 

I shall club them all and also add the `.into_report` at the end. Let me know if that is not as intended.